gms/gossiper.cc

```diff
@@ -1701,13 +1729,42 @@ void gossiper::apply_new_states(inet_address addr, endpoint_state& local_state,
     utils::chunked_vector<application_state> changed;
     auto&& remote_map = remote_state.get_application_state_map();
 
+    bool replicate_failed = false;
+    bool listener_failed = false;
+
+    auto isolate_node = [&] {
+        try {
+            logger.warn("Node {} is isolated by gossip because it failed to run gossip listener. A rolling restart is needed to recover!", addr);
+            _isolated_nodes_due_to_listener_failure.insert(addr);
+            container().invoke_on_others([addr] (gossiper& g) {
+                g._isolated_nodes_due_to_listener_failure.insert(addr);
+            }).get();
+            logger.info("Mark node {} as DOWN because it is isloated", addr);
+            convict(addr);
+        } catch (...) {
+            logger.error("Failed to isolate node {}", addr, std::current_exception());
```

Comment:
If we failed to apply a gossip update, we're likely to fail here too.

Reply:
Not necessarily, the failure can be due to listener-specific logic. That's the case which motivated this patch. See https://github.com/scylladb/scylla/pull/9658#discussion_r765358252

Reply:
Yes.